### PR TITLE
perf: finish getting rid of uses of the Roboto font

### LINKED ISSUE #26589: Remove Roboto Font to Improve Performance and Consistency
### What is this about?

This task involves the removal of the Roboto font from the MetaMask extension. Currently, Roboto is being used as a fallback font, but it is mostly unnecessary since Euclid Circular B is the primary font. The investigation has revealed that Roboto is only utilized in a few select instances and could likely be removed entirely to streamline the font loading process and improve performance.

### Scenario

_No response_

### Design

_No response_

### Technical Details

- Audit the extension to confirm all instances of Roboto usage.
- Replace Roboto with Euclid Circular B where necessary.
- If Euclid is unavailable, configure the font stack to fall back to system fonts rather than another custom font.
- Ensure that the removal does not affect any Material UI components or external packages.

### Threat Modeling Framework

_No response_

### Acceptance Criteria

- Roboto should be completely removed from the MetaMask extension's CSS and other relevant files.
- The font stack should fall back to system fonts when Euclid Circular B is unavailable.
- Verify that the removal does not negatively impact the UI/UX.
- Test to confirm no regression or issues arise from this change.

### Stakeholder review needed before the work gets merged

- [X] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

- [Slack thread](https://consensys.slack.com/archives/CTQAGKY5V/p1724173426966159)

### PR DESCRIPTION
**Pull request by davidmurdoch**
### Description

We use two custom fonts in our typography CSS: Euclid Circular B and Roboto.

Roboto is a fallback font, but since both are custom fonts Roboto will never be activated by our default typography. It can be removed.

Roboto is only used in one other place in our CSS, but the selector looks to be not used anymore.

Lastly, Material UI uses Roboto by default for input elements and their labels. This PR updates our use of it to use "Euclid Circular B" and our usual fall back fonts. I don't particularly like the implementation (setting `fontFamily` as a string constant in JS), but I'm not sure how to do it any better (cc @georgewrmarshall ).

It saves about 0-2ms from the load time (compared to `popup-defer-scripts` branch), so it's not substantial by any stretch. It does save 1.4 megabytes, about 6%, from the dist zip (compared to `popup-defer-scripts` branch), which seems worthwhile.

Fixes: https://github.com/MetaMask/metamask-extension/issues/26589

### CODE REVIEW

File: ui/components/ui/text-field/text-field.component.js [hunk 1]
```diff
@@ -19,6 +19,13 @@ const inputLabelBase = {
   color: 'var(--color-text-default)',
 };
 
+const fontFamily = [
+  '"Euclid Circular B"',
+  'Helvetica',
+  'Arial',
+  'sans-serif',
+].join(', ');
+
```
Comment (davidmurdoch) [inline ui/components/ui/text-field/text-field.component.js:28]: i hate this. is there a better way to get our design-system's font family?

File: ui/pages/swaps/fee-card/index.scss [hunk 2]
```diff
@@ -207,11 +207,6 @@
   &__bold {
     font-weight: bold;
   }
-
-  &__tilde {
```
Comment (davidmurdoch) [inline ui/pages/swaps/fee-card/index.scss:211]: I couldn't find any use of this class anywhere in our compiled code.

File: ui/components/ui/text-field/text-field.component.js (hunk 1 above)
Comment (georgewrmarshall) [reply ui/components/ui/text-field/text-field.component.js:28]: Unfortunately, not really. We can use the typography object from the design tokens package, but it's still pretty janky. 

```javascript
import { typography } from '@metamask/design-tokens';

const fontFamily = [
  typography.fontFamilies.euclidCircularB,
  'Helvetica',
  'Arial',
  'sans-serif',
].join(', ');
```

**Review by georgewrmarshall — APPROVED** [review]
Excellent work thank you for this 💯 LGTM!
- Checked `TextField` label in storybook to confirm no visual regressions ✅ 
- Checked no instances of `Roboto` remain in codebase

Screenshot 2024-08-21 at 2 43 41 PM

Screenshot 2024-08-21 at 2 44 17 PM

**Review by matthewwalsh0 — APPROVED** [review]

**Review by vthomas13 — APPROVED** [review]

**Review by jclancy93 — APPROVED** [review]

**Review by NidhiKJha — APPROVED** [review]

**Review by darkwing — APPROVED** [review]

**Pull request merged**